### PR TITLE
MVP1 Design Changes

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

Replaces the old layout/designs with a new layout with the template and designs defined in the [figma file](https://www.figma.com/file/ZDPonnBXbMhV2smf51Dylc/?node-id=3628%3A102523), and updates all the pages that used the old layout (`Main`) to use the new one (`MainLayout`).

Most of the design work was centered around the [`MainLayout`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/layouts/Main.tsx) component, which is used to structure a given page with the new design. It defines the grid layout, spacings, and has props to set the title/heading, as well as any success or error messages using the new `Summary` components. Content that goes into this layout is done using `children`.

Alongside the `MainLayout`, some helper components were created to simplify how to layout the `children` to the layout.
Specifically the `MainBodyText` and `MainForm` (+ related `PasswordFormMainLayout`) components.

[`MainBodyText`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/components/MainBodyText.tsx) is the component to use to add text/paragraphs corresponding to the designs in this layout.

[`MainForm`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/components/MainForm.tsx) is used to wrap all forms that are used on the pages using the `MainLayout` and helps to simplify form creation. By default it includes the `CsrfFormField`, a customizable submit button (through props), as well as the optional ability to add the terms and conditions for the Guardian and Recaptcha separately. Input fields are passed as `children` to this component, and will automatically be picked up as a form field. At the very least a `formAction` (url to `POST` the form to), and `submitButtonText` must be provided.
In the future when implementing recaptcha, the `useRecaptcha` flag could be used to easily enable/disable recaptcha for a given form on a page.

Once those components (and respective stories) were in place much of the rest of the diffs in this PR is the refactoring of all of the pages from the old layout to the new layout.

One thing for the future is to combine the `MainGrid` layout (used for the sign in/register pages) with the new `MainLayout` component now used for many of the other pages.

## How to test/review this PR

- Open the [figma file](https://www.figma.com/file/ZDPonnBXbMhV2smf51Dylc/?node-id=3628%3A102523) on the `✅ -  Check your inbox - refinement` page to see expected designs
- Open the [latest storybook](https://60929d168594f80039336501-plgtaqvjar.chromatic.com/) for this branch to see new designs
- Open the [latest chromatic UI tests](https://www.chromatic.com/build?appId=60929d168594f80039336501&number=1416) to see what's changed

### CODE REVIEW

File: src/client/components/Container.tsx [hunk 1]
```diff
@@ -30,9 +30,13 @@ const noPaddingStyles = css`
...
-const sideBorderColors = (color: string) => css`
-  border-left-color: ${color} !important;
-  border-right-color: ${color};
+const sideBorderStyles = (color = neutral[86]) => css`
+  > div {
+    ${from.tablet} {
+      border-left: 1px solid ${color};
+      border-right: 1px solid ${color};
+    }
+  }
```
Comment (coldlink) [inline src/client/components/Container.tsx:39]: Fixes border alignment on the Container component used by the navigation bar, footer, and layouts.

File: src/shared/model/QueryParams.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export interface QueryParams
   extends PersistableQueryParams,
     StringifiableRecord {
   emailVerified?: boolean;
+  emailSentPage?: boolean;
```
Comment (coldlink) [inline src/shared/model/QueryParams.ts:31]: This query parameter is used to determine if the request from sent from an email sent page, for example when resending an email. We use this parameter to determine whether to show or hide the "Email sent" success message that appears when you resend an email.

File: src/client/layouts/Main.tsx [hunk 3]
```diff
@@ -1,54 +1,162 @@
...
+              message={successMessage}
+            />
+          )}
+          {pageTitle && (
+            <header css={headerStyles(!!(errorMessage || successMessage))}>
+              <h1 css={[pageTitleStyles]}>{pageTitle}</h1>
+            </header>
+          )}
+          <div
+            css={bodyStyles(!!(pageTitle || successMessage || errorMessage))}
```
Comment (ob6160) [inline src/client/layouts/Main.tsx:153]: the cases where `!!(...)` is happening for `bodyStyles`, `headerStyles` etc could maybe benefit from being moved into variables for clarity

File: src/client/pages/ChangePassword.tsx [hunk 4]
```diff
@@ -24,24 +19,13 @@ export const ChangePassword = ({
...
-            submitButtonText={buttonText}
-          />
-        </PageBody>
-      </PageBox>
-    </Main>
-    <Footer />
-  </>
+  <MainLayout pageTitle={headerText}>
+    <MainBodyText>Please enter your new password for {email}</MainBodyText>
+    <PasswordFormMainLayout
```
Comment (ob6160) [inline src/client/pages/ChangePassword.tsx:24]: way cleaner now using this, nice job :)

File: src/client/pages/EmailSent.tsx [hunk 5]
```diff
@@ -1,99 +1,104 @@
...
+      </MainBodyText>
+      {previousPage && (
+        <MainBodyText>
+          Wrong email address?{' '}
+          <Link subdued href={previousPage}>
+            Change email address
+          </Link>
+        </MainBodyText>
+      )}
+      {email && resendEmailAction && hasJS && (
```
Comment (ob6160) [inline src/client/pages/EmailSent.tsx:66]: just wondering if we still need to check for client side JS with `hasJS` on this one now that the `{loading ? 'Resending...' : 'Resend email'}` bit is gone?

File: src/client/pages/EmailSentPage.tsx [hunk 6]
```diff
@@ -6,17 +6,24 @@ import { addQueryParamsToPath } from '@/shared/lib/queryParams';
...
+    emailSentPage: true,
+  });
 
   return (
     <EmailSent
       email={email}
       previousPage={previousPage}
       resendEmailAction={previousPage}
       queryString={queryString}
+      showSuccess={!!emailSentPage}
```
Comment (ob6160) [inline src/client/pages/EmailSentPage.tsx:24]: just a more general question: what does `emailSentPage` represent here to indicate success? (I'm sure it does, but it's not clear in my head)

File: src/client/pages/ResetPassword.tsx [hunk 7]
```diff
@@ -26,45 +28,45 @@ export const ResetPassword = ({
   buttonText,
   queryString = '',
   formActionOverride,
+  inputLabel,
```
Comment (ob6160) [inline src/client/pages/ResetPassword.tsx:31]: small one, but could be worth being specific around the name `inputLabel` as it seems to be mainly used as an `emailInputLabel` across the pages 🤔

File: src/server/lib/queryParams.ts [hunk 8]
```diff
@@ -3,12 +3,12 @@ import { validateReturnUrl, validateRefUrl } from '@/server/lib/validateUrl';
 import { validateClientId } from '@/server/lib/validateClientId';
 import { FederationErrors } from '@/shared/model/Errors';
 
-const validateEmailVerified = (emailVerified?: string): boolean | undefined => {
-  if (!emailVerified) {
+const validateBoolean = (maybeBoolean?: string): boolean | undefined => {
```
Comment (ob6160) [inline src/server/lib/queryParams.ts:6]: could this one be something like `isStringTrue` maybe? had to read the function to work out what it did, although it was already quite ambiguous before the name change lol

File: src/shared/model/QueryParams.ts (hunk 2 above)
Comment (ob6160) [reply src/shared/model/QueryParams.ts:31]: ah, I left a comment above before seeing your comment here. I think I wasn't sure what the meaning was until I read your comment/how it was used. Maybe a name like `fromEmailSentPage` or `emailSentSuccess` could be more self explanatory 🤔

**Review by ob6160 — COMMENTED** [review]
great work! a lot of nice changes in here 🥇🥇  I especially like the `<MainForm>` and `<Main>`  abstractions. I have left some comments for perusal at your leisure

File: src/shared/model/QueryParams.ts (hunk 2 above)
Comment (coldlink) [reply src/shared/model/QueryParams.ts:31]: I like `emailSentSuccess`! I'll change it to that.

File: src/client/components/PasswordForm.tsx [hunk 9]
```diff
@@ -147,7 +148,7 @@ const Weak = () => {
   `;
 
   return (
-    <div css={[messageWrapperStyles, baseMessageStyles]}>
+    <div css={[noMargin ? undefined : messageWrapperStyles, baseMessageStyles]}>
```
[the remark this answers is not in the document]
Comment (coldlink) [reply src/client/components/PasswordForm.tsx:151]: It's difficult, but emotion is okay with `undefined` values in an array, which is why I used a ternary operator to do this, as emotion ignores any undefined values.

[the remark this answers is not in the document]
Comment (coldlink) [reply src/client/components/PasswordForm.tsx:151]: `&&` or `||` don't work as emotion doesn't like `boolean` values 😅

File: src/client/components/Terms.tsx [hunk 10]
```diff
@@ -10,10 +10,7 @@ const Text = ({ children }: { children: React.ReactNode }) => (
     css={css`
       ${textSans.xxsmall()}
       margin-top: 0;
-      margin-bottom: 0px;
```
[the remark this answers is not in the document]
Comment (coldlink) [reply src/client/components/Terms.tsx:13]: Yeah, there should always be a 6px margin bottom between the bottom of the input field and the terms text.

File: src/client/pages/EmailSent.tsx (hunk 5 above)
Comment (coldlink) [reply src/client/pages/EmailSent.tsx:68]: you're right! it's not needed anymore, so I'll remove it.

File: src/server/lib/queryParams.ts (hunk 8 above)
Comment (coldlink) [reply src/server/lib/queryParams.ts]: i've changed it to `isStringBoolean`!

File: src/client/components/PasswordForm.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (ob6160) [reply src/client/components/PasswordForm.tsx:151]: sounds good!

**Review by ob6160 — APPROVED** [review]
left a few comments on Chromatic after going through all the diffs on there

**Comment by coldlink** [comment]
> left a few comments on Chromatic after going through all the diffs on there

I actually forgot those pages were a thing. It was for an AB test that we ran. The pages are no longer used or required, so we should really just remove the dead code. I'll try and raise a separate PR for this.

**Pull request merged**